src/InputInterface.php

```diff
@@ -5,12 +5,20 @@
 
 /**
  * Interface representing an input element
+ *
+ * @template TInput of NodeInterface
  */
 interface InputInterface extends NodeInterface
 {
-    public function setName(string $name): InputInterface;
+    /**
+     * @return TInput
+     */
+    public function setName(string $name);
```

Comment:
I don't understand why remove the return type, setName and setValue always return `$this`.

Reply:
This is php generics: https://medium.com/@ondrejmirtes/generics-in-php-using-phpdocs-14e7301953

Reply:
this way you don't have to say _It is an interface_ when it really it is the implementation of the interface